src/internal/Thresholding.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file Thresholding.cpp
+ * @author Luciana Moreno (lumoreno.borges@gmail.com)
+ * @brief Thresholding class
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <bits/stdc++.h>
+#include <chrono>
+#include <hyro/Thresholding.h>
+
+namespace hyro
+{
+  /**
+   * @brief Construct a new Thresholding:: Thresholding object
+   */
+Thresholding::Thresholding(double data_sample, double amplitude):
+  data_sample(data_sample), amplitude(amplitude)
+{
+}
+
+/* Convert the signal value in a digital one */
```

Comment:
According to the exercise, this class should be able to receive two arguments, **threshold** and **amplitude**.

I would do different. I would change the function `makeThresholding()` to `getThreshValue(float signal)` and implemented it like as follows:

```c++
return (signal < threshold) ?  0 : m_amplitude
```

Reply:
I should use a parameter `threshold` instead of a constant called **THRESHOLDING**.